Dockerfile

```diff
@@ -158,119 +203,35 @@ ENV \
-
-sed -i 's/api\.versionsFrom/\/\/api.versionsFrom/' /home/wekan/app/packages/meteor-useraccounts-core/package.js
-cd /home/wekan/.meteor
-gosu wekan:wekan /home/wekan/.meteor/meteor -- help
-
-# Build app (Production)
-cd /home/wekan/app
-mkdir -p /home/wekan/.npm
-chown --recursive wekan:wekan /home/wekan/.npm
-chmod u+w *.json
-gosu wekan:wekan meteor npm install --production
-gosu wekan:wekan /home/wekan/.meteor/meteor build --directory /home/wekan/app_build
-cd /home/wekan/app_build/bundle/programs/server/
-chmod u+w *.json
-gosu wekan:wekan meteor npm install --production
-cd node_modules/fibers
-node build.js
-cd ../..
-# Remove legacy webbroser bundle, so that Wekan works also at Android Firefox, iOS Safari, etc.
-rm -rf /home/wekan/app_build/bundle/programs/web.browser.legacy
-mv /home/wekan/app_build/bundle /build
-
-# Put back the original tar
-mv $(which tar)~ $(which tar)
-
-# Cleanup
-apt-get remove --purge --assume-yes ${BUILD_DEPS}
-npm uninstall -g api2html
-apt-get autoremove --assume-yes
-apt-get clean --assume-yes
-rm -Rf /tmp/*
-rm -Rf /var/lib/apt/lists/*
-rm -Rf /var/cache/apt
-rm -Rf /var/lib/apt/lists
-rm -Rf /home/wekan/app_build
-rm -Rf /home/wekan/app
-rm -Rf /home/wekan/.meteor
-
-mkdir /data
-chown wekan --recursive /data
-EOR
-
+# Install node and application sources from build stage
+RUN --mount=type=bind,from=downloads,source=node.tar.gz,target=node.tar.gz \
+    --mount=type=bind,from=build,source=/build,target=/build \
+    # Install NodeJS
+    tar xzf node.tar.gz --strip-components=1 --keep-old-files --no-same-owner && \
+    # Add the wekan user
+    useradd --uid 999 --user-group --system wekan && \
+    # Copy wekan bundle
+    cp -r /build/bundle /wekan && \
+    # Remove the web.browser.legacy platform
+    rm -r /wekan/programs/web.browser.legacy && \
+    # Change ownership of wekan directory
+    chown wekan:wekan -R /wekan
+
+# Change to the wekan user
 USER wekan
 
+# Set the port to listen on
 ENV PORT=8080
-EXPOSE $PORT
+EXPOSE ${PORT}
+
+# Set the maximum stack size
+ENV STACK_SIZE=65500
+
+# Define the stop signal for node
+STOPSIGNAL SIGINT
 
-STOPSIGNAL SIGKILL
-WORKDIR /home/wekan/app
+# Declare a volume on /data
+VOLUME /data
 
-#---------------------------------------------------------------------
-# https://github.com/wekan/wekan/issues/3585#issuecomment-1021522132
-# Add more Node heap:
-#   NODE_OPTIONS="--max_old_space_size=4096"
-# Add more stack:
-#   bash -c "ulimit -s 65500; exec node --stack-size=65500 main.js"
-#---------------------------------------------------------------------
-#
-# CMD ["node", "/build/main.js"]
-# CMD ["bash", "-c", "ulimit -s 65500; exec node --stack-size=65500 /build/main.js"]
-# CMD ["bash", "-c", "ulimit -s 65500; exec node --stack-size=65500 --max-old-space-size=8192 /build/main.js"]
-CMD ["bash", "-c", "ulimit -s 65500; exec node /build/main.js"]
+# Set startup command
+CMD ["bash", "-c", "ulimit -s ${STACK_SIZE} && exec node /wekan/main.js"]
```

Comment:
This is the problem 2, to be removed:
```
-s ${STACK_SIZE}
```

Reply:
https://github.com/wekan/wekan/blob/main/Dockerfile#L276 does `-s 65500`.

Reply:
I think it should be about The maximum number of open file descriptors. Not about stack size.
```
ulimit -n 65500
```
https://ss64.com/bash/ulimit.html

Reply:
Furthermore, If you want `-s $(STACK_SIZE}` removed, it would mean you want `ulimit && node /wekan/main.js`?
This does not do anything at all, as `ulimit` without parameters just displays the current `ulimit` (see https://ss64.com/bash/ulimit.html):
> If limit is given, it is the new value of the specified resource. Otherwise, the current value of the soft limit for the specified resource is printed, unless the '-H' option is supplied.

Would you like me to change to:
```dockerfile
CMD [ "node", "/wekan/main.js" ]
```
?

Reply:
> I think it should be about The maximum number of open file descriptors. Not about stack size.
> 
> ```
> ulimit -n 65500
> ```
> 
> https://ss64.com/bash/ulimit.html

I see, so there is a mistake on `main`.
Should l make the change?
It seems to me like it is not required, since it did not exist in `main`, and we'd just be better off with `CMD [ "node", "/wekan/main.js" ]`.

Reply:
Stack size setting removed, and only this:
```
CMD ["bash", "-c", "ulimit -n 65500 && exec node /wekan/main.js"]
```

Reply:
Acknowledged. Making the changes now.

Reply:
@NadavTasher 

Original reasons are these, and I still am not sure about what would be correct settings:
- https://github.com/wekan/wekan/issues/3585#issuecomment-1021522132
- https://stackoverflow.com/questions/20936486/node-js-maximum-call-stack-size-exceeded
- https://github.com/wekan/wekan/issues/5347

Setting stack too high causes too much RAM usage, that causes server to start swapping.

Also, because of memory leaks https://github.com/wekan/wekan/issues/5140 there is a very bad workaround:

https://github.com/wekan/wekan/wiki/Cron

Reply:
@xet7 

These two problems do not seem to have any relation to maximim FDs

Reply:
Ok, these are not related to FDs.

Reply:
I will change the Dockerfile.